operator/api.go

```diff
@@ -112,11 +112,17 @@ func healthHandler(w http.ResponseWriter, r *http.Request) {
 // checkStatus checks the connection status to the kvstore and
 // k8s apiserver and returns an error if any of them is unhealthy
 func checkStatus() error {
-	if kvstoreEnabled() {
-		if client := kvstore.Client(); client == nil {
-			return fmt.Errorf("kvstore client not configured")
-		} else if _, err := client.Status(); err != nil {
-			return err
+	// We check for leader status here because only the leader has access to
+	// the kvstore client. Otherwise, the check for the kvstore client will
+	// block. It is safe for a non-leader to skip this status check, as the
+	// leader will report the status of the kvstore client.
+	if leader := isLeader.Load().(bool); leader {
```

Comment:
Can we always safely assume that Store() has been called before (else it'll panic)?
I wonder if it make sense to use a safer version - https://play.golang.org/p/Qt7cQ87GPom

Reply:
Fixed.